case-lib/hijack.sh

```diff
@@ -64,13 +64,10 @@ function func_exit_handler()
 
     # get ps command result as list
     local -a cmd_lst
-    # $$ as current script pid
-    # NOTICE: already test with $BASHPID:
-    # it can output the same result of $$
-    # but the result could not be stored in the array
-    readarray -t cmd_lst < <(pgrep -P $$ -a|grep -v "$SCRIPT_NAME")
```

Comment:
I checked `pgrep -P $$ -a` it will show the script it self, how do you remove it from the list

Reply:
Wrong, demo:
```shell
hostname:~$ pgrep -P $$ -a
hostname:~$ sleep 1000 &
[1] 140297
hostname:~$ 
hostname:~$ ps f
    PID TTY      STAT   TIME COMMAND
 140238 pts/3    Ss     0:00 -bash
 140297 pts/3    S      0:00  \_ sleep 1000
 140298 pts/3    R+     0:00  \_ ps f
 139197 pts/2    Ss+    0:00 -bash
 138772 pts/1    Ss+    0:00 /bin/sh -i
hostname:~$ echo $$
140238
hostname:~$ pgrep -P $$ -a
140297 sleep 1000
```